applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/servers.js

```diff
@@ -1,21 +1,59 @@
 'use strict';
 'require form';
+'require uci';
 'require shadowsocks-libev as ss';
 
-function startsWith(str, search) {
-	return str.substring(0, search.length) === search;
-}
+var conf = 'shadowsocks-libev';
 
 return L.view.extend({
 	render: function() {
 		var m, s, o;
 
-		m = new form.Map('shadowsocks-libev', _('Remote Servers'),
+		m = new form.Map(conf, _('Remote Servers'),
 			_('Definition of remote shadowsocks servers.  \
 				Disable any of them will also disable instances referring to it.'));
 
 		s = m.section(form.GridSection, 'server');
 		s.addremove = true;
+		s.renderSectionAdd = function(extra_class) {
+			var el = form.GridSection.prototype.renderSectionAdd.apply(this, arguments);
+			el.appendChild(E('button', {
+				'class': 'cbi-button cbi-button-add',
+				'title': _('Import Links'),
+				'click': function() {
```

Comment:
This is a personal taste matter, but I usually move complex event handler functions into separate `handleXYZ` methods which I then invoke. To do that, you need to preserve the `this` context though.

```
s.renderSectionAdd = L.bind(function(extra_class) {
    ...
}, this);
```

And later within the `el.appendChild()`:

```
E('button', {
    ...
    'click': ui.createHandlerFn(this, 'handleLinkImport')
}, [ _('Import Links') ]);
```

Then move this anonymous function code to a `handleLinkImport` view class method.